provider/transformation.go

```diff
@@ -0,0 +1,209 @@
+package provider
+
+import (
+	"fmt"
+)
+
+type (
+	OutTransformationProvider[T any] interface {
+		convertToInt(T) (int64, error)
+		convertToString(T) (string, error)
+		convertToFloat(T) (float64, error)
+		convertToBool(T) (bool, error)
+		outTransformations() []OutTransformation
+	}
+	InTransformationProvider interface {
+		setParam(string, any) error
+		inTransformations() []InTransformation
+	}
+	ScriptTransformationProvider[A any] interface {
+		evaluate() (A, error)
+	}
+)
+
+type TransformationConfig struct {
+	Name, Type string
+	Config     Config
+}
+
+type InTransformation struct {
+	name     string
+	function func() (any, error)
+}
+
+type OutTransformation struct {
+	name, Type string
```

Comment:
must `Type` be public?

Reply:
No, but lower case `type` is obviously a reserved word. I have changed it to `valueType` for now, but maybe there is a better naming.